common/src/main/java/org/conscrypt/HpkeSuite.java

```diff
@@ -0,0 +1,330 @@
+      this.pkLength = pkLength;
+      this.skLength = skLength;
+    }
+
+    /**
+     * KEM id in its decimal representation
+     *
+     * @return kem id
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">KEM
+     *         ids</a>
+     */
+    int getId() {
+      return id;
+    }
+
+    /**
+     * The length in bytes of an encapsulated key produced by this KEM.
+     *
+     * @return encapsulated key size in bytes
+     */
+    int getEncLength() {
+      return encLength;
+    }
+
+    /**
+     * The length in bytes of an encoded public key for this KEM.
+     *
+     * @return public key size in bytes
+     */
+    int getPkLength() {
+      return pkLength;
+    }
+
+    /**
+     * The length in bytes of an encoded private key for this KEM.
+     *
+     * @return private key size in bytes
+     */
+    int getSkLength() {
+      return skLength;
+    }
+
+    /**
+     * Validates the enc size in bytes matches the {@link KEM} spec.
+     *
+     * @param enc encapsulated key produced by the kem
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">expected
+     *         enc size</a>
+     */
+    void validateEnc(byte[] enc) {
+      Preconditions.checkNotNull(enc, "enc");
+      final int expectedLength = this.getEncLength();
+      if (enc.length != expectedLength) {
+        throw new IllegalArgumentException(
+            "Expected enc length of " + expectedLength + ", but was " + enc.length);
+      }
+    }
+
+    /**
+     * Validates the public key size in bytes to match the {@link KEM} public key spec.
+     *
+     * @param publicKey alias pk
+     * @return key in its raw format
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-algorithm-identifiers">expected
+     *         pk size</a>
+     */
+    byte[] validateAndGetPublicKey(PublicKey publicKey) {
+      Preconditions.checkNotNull(publicKey, "publicKey");
+      if (!(publicKey instanceof OpenSSLX25519PublicKey)) {
+        throw new IllegalArgumentException(
+            "Public key algorithm " + publicKey.getAlgorithm() + " is not supported");
+      }
+
+      final byte[] key = ((OpenSSLX25519PublicKey) publicKey).getU();
+      final int expectedLength = this.getPkLength();
+      if (key.length != expectedLength) {
+        throw new IllegalArgumentException("Expected public key length of " + expectedLength
+            + ", but was " + key.length);
+      }
+      return key;
+    }
+
+    /**
+     * Validates the private key size in bytes to match the {@link KEM} private key spec.
+     *
+     * @param privateKey alias sk
+     * @return key in its raw format
+     * @see <a
+     *         href="https://www.rfc-editor.org/rfc/rfc9180.html#name-algorithm-identifiers">expected
+     *         sk size</a>
+     */
+    byte[] validateAndGetPrivateKey(PrivateKey privateKey) {
```

Comment:
Ditto that this isn't (always) all the validation steps. It is for X25519, but not for other KEMs.

Reply:
Same as above. I have updated the name to perform validations only on the length, and updated the name to reflect the raw private key will be extracted from the private key object.